src/iop/demosaic.c

```diff
@@ -1200,6 +1200,32 @@ static void vng_interpolate(float *out, const float *const in, const dt_iop_roi_
     for(int i = 0; i < height * width; i++) out[i * 4 + 1] = (out[i * 4 + 1] + out[i * 4 + 3]) / 2.0f;
 }
 
+/** 1:1 "demosaic" where in comes RGB buffer and out goes R=G=B=average(R,G,G,B) */
+static void rgb_monochrome(float *out, const float *const in, const dt_iop_roi_t *const roi_out,
+                           const dt_iop_roi_t *const roi_in)
+{
+  // we should always process buffers of exactly the same size
+  assert(roi_in->width == roi_out->width);
+  assert(roi_in->height == roi_out->height);
+
+#ifdef _OPENMP
+#pragma omp parallel for default(none) shared(out) schedule(static)
+#endif
+  for(int row = 0; row < roi_out->height; row++)
+  {
+    for(int col = 0; col < roi_out->width; col++)
+    {
+      int pos = 4*(row*roi_in->width+col);
+      // Average four pixels giving green twice the weight because of RGGB
+      float avg = (in[pos] + (in[pos+1] * 2.0f) + in[pos+2]) / 4.0f;
+
+      // Set RGB all to the average
+      pos = 4*(row*roi_out->width+col);
```

Comment:
That may be needed if the input size isn't always the same as the output. I had assumed (and asserted) that was the case but maybe it's not?